### DISCUSSION TITLE
Diagnostic- Bad Control Port & 6581

### DISCUSSION THREAD

**Discussion by DariBer** [frntc/SIDKick-pico#67] [Q&A]
I have a SIDKick Pico 0.2 board with firmware 0.202. 

When I run a dignostic (586220) I get all fine with a MOS 6581 chip, but when I switch to SK Pico I get BAD Control Port and BAD 6581. 

I have switched to 6581 in the SK Pico menu. 

Is this normal?

Bellow is the SK Pico settings and from Diagnostic

![image](https://github.com/user-attachments/assets/9deaa360-7494-4063-bdfb-f7d71adf6776)

![image](https://github.com/user-attachments/assets/a7fd7668-3a86-445c-881e-9a57e5bd03a9)

**Comment by frntc** [comment]
This might be due to paddle values being a bit off with the level shifting. Could you run this Basic program and let me know the numbers it outputs when the joystick-port harness is connected?

10 PRINT PEEK(54297)
20 GOTO 10

**Reply by DariBer** [reply]
The number it prints on the screen is 255

**Reply by frntc** [reply]
A value of 255 means that there is no paddle/resistor connected to the POTX/Y pins. Did you connect to both joystick ports? Do paddles/mouse work otherwise? (e.g. test with https://csdb.dk/release/?id=179703)

**Reply by DariBer** [reply]
When I test the joystick and pads with MOS 6581 it works fine and after I run you little basic program, it switches between 91 and 92.

When I switch to SIDKick Pico and connect at joystic, it works fine, but when I connect a pad then it acts weird, when I move the wheel it only moves a smal range, not the whole, and on the other pad it moves all over the place. and your basic program shows 255.

I hear the sound fine when I use the SK Pico, so it's the pads that seems to be the problem. I haven't changed anything in the SK Pico config file for Pot X/Y as you can see in the program above.

**Reply by frntc** [reply]
Are you using a Pico-clone? Have you read about paddles here: https://github.com/frntc/SIDKick-pico#troubleshooting ?

**Reply by DariBer** [reply]
I bought a kit from retro8bitshop.com, and in that kit it was a black Pico clone. I have original Pico I could try to see if there is any difference. 

Will read the text now.

**Reply by DariBer** [reply]
I installed an original Pico and installed the firmware 0.202. 

Still get BAD at Control and 6581

Now when I run your basic program it gives me numbers from 71 to 76, so little bit lower than original SID. 

But I saw another weird thing, when one pad is in top 255, the other also goes to 255. See attached video. 

https://github.com/user-attachments/assets/11d8ca09-4cd0-4f65-bffb-c7c56593cb9b

Forgot to say, it's an Assy 250425 board.

**Reply by frntc** [reply]
Afaik the diag ROMs test the paddle values for the 80 to 120 range, so that values you get with the harness are a tiny bit too low (but that doesn't mean that paddles don't work). Maybe aging caps etc on the C64 boards also influence the interplay with the SKpico. The other problem (255) seems weird, have you measured the potentiometer resistances of your paddles?

**Reply by DariBer** [reply]
I will look more into the wired 255 problem. Se if I get same problem with original SID. 

So there is no way to change configuration so that the SK Pico get's values between 80 and 120?

Change any resistor on the board?

**Reply by DariBer** [reply]
>The other problem (255) seems weird, have you measured the potentiometer resistances of your paddles?

@frntc the Commodore paddles work fine when I use MOS 6581. I have cleaned the potentiometers and the paddles works fine. To me it seems to be something with the SIDKick Pico. I have tried to put the SIDKick Pico in 2 different Commodore 64, both are Assy 250425. They behave weird, when I put one of them max to the right, then the other one also goes to 255. 

Do you think this is a firmware or hw issue?

I'm using a Commodore Paddles that are 2 pad in one port

<img width="492" alt="Screenshot 2024-08-13 at 16 51 59" src="https://github.com/user-attachments/assets/d081d08f-30e5-4504-b050-057cf0c57ec0">

**Reply by frntc** [reply]
Are you using a Pico clone? Checked that the ADC-reference voltage is correct? (see troubleshooting in the readme)

**Reply by DariBer** [reply]
it was Pico clone, but in an earlier post you told me to look at the troubleshooting section. after that I changed to a RPi Pico.
On the Pico (PIN35, ADC_VREF) I get 3.325V, is that the ADC voltage you asked for?

**Reply by frntc** [reply]
then it should be fine (too many things, so I overlook what I have already answered :)). No idea yet though about the 255-issue. Could be many things. I haven't experienced nor heard about that previously. If paddles didn't work in my Commodore machines, it was always the 4066 (in case you haven't checked it -- or did I already mention that before? :))

**Reply by DariBer** [reply]
Could be the 4066, but.... everything works fine if I put an original MOS 6581. When I remove the MOS 6581 and install the SK Pico (HW 0.2, firmware 0.202) then I get the errors. 

Should I try an older firmware to see if I get same problem?

**Reply by DariBer** [reply]
The problem I have now is as in the Video, it jumped all over when one of the paddles are to the right, value 255, then the other is all over and also move to 255 without touching that paddle.

**Reply by DariBer** [reply]
@frntc I'm not at not at home now so can't test the new firmware. 

But do you think the new firmware can solve any of the "problem" above. 

That it fails when doing a diagnostic test?

The other one is that paddle jumps around or both go to max right (255). If one of them is turned to max, the other one goes automatically also to max

**Reply by frntc** [reply]
not sure :-) I don't have an explanation for the behavior that you describe (maybe I had one if I'd see oscilloscope measurements). Have you had a chance to test with another 4066?

**Reply by DariBer** [reply]
I didn't have any 4066 at home, So waiting for them to arrive. 

I can send you info from oscilloscope measurements. 

Where do you want me to measure, what IC and what PINs would like me to look for?

**Reply by frntc** [reply]
Pin 16 and 17 on the Pico (HW rev 2, right?) with paddles set to "still working" and then "failing". One should see several measurement cycles (each is approx. 500uS). Thanks!

**Reply by DariBer** [reply]
OK, will do that, if I don't do it today or tomorrow I will definitely then do it at the weekend
